components/AddTransactionModal.tsx

```diff
@@ -0,0 +1,313 @@
+                                type="number"
+                                id="quantity"
+                                placeholder="0.00"
+                                onChange={(e) =>
+                                    setQuantity(Number(e.target.value))
+                                }
+                            />
+                        </div>
+                        <div className="max-w-fit items-center gap-1.5">
+                            <Label htmlFor="price">Stock Price</Label>
+                            <Input
+                                type="number"
+                                id="price"
+                                onChange={(e) =>
+                                    setStockPrice(Number(e.target.value))
+                                }
+                            />
+                        </div>
+                    </div>
+
+                    <div className="mt-4 flex flex-col justify-start gap-1.5">
+                        {/* TODO: API call for each change in purchase date to find the price at the chosen purchase date */}
+                        <Popover>
+                            <Label htmlFor="date-picker">Purchase Date</Label>
+                            <PopoverTrigger asChild>
+                                <Button
+                                    variant={"outline"}
+                                    className={cn(
+                                        "w-full justify-start text-left font-normal",
+                                        !date && "text-muted-foreground",
+                                    )}
+                                    id="date-picker"
+                                >
+                                    <CalendarIcon className="mr-2 h-4 w-4" />
+                                    {date ? (
+                                        format(date, "PPP")
+                                    ) : (
+                                        <span>Pick a date</span>
+                                    )}
+                                </Button>
+                            </PopoverTrigger>
+                            <PopoverContent className="w-auto p-0">
+                                <Calendar
+                                    mode="single"
+                                    selected={date}
+                                    onSelect={setDate}
+                                    initialFocus
+                                />
+                            </PopoverContent>
+                        </Popover>
+                    </div>
+
+                    <div className="mt-4 h-[75px] w-full rounded-xl bg-gray-200">
+                        <div className="grid w-full max-w-sm items-center gap-1.5 p-4">
+                            <Label htmlFor="spent">Total Spent</Label>
+                            <p id="spent" className="font-bold text-xl">
+                                ${(quantity * stockPrice).toFixed(2)}
```

Comment:
can use the [`formatUsd(quantity * stockPrice)` funtion](https://github.com/is442oop/portfolio-analyzer-web-ui/blob/main/utils/functions.ts#L1) like:

```ts
import { formatUsd } from "@/utils/functions";

const quantity = 10.0
const stockPrice = 25000.0;

console.log(formatUsd(quantity * stockPrice));  // prints '$250,000.00'
```

example use case: [@/components/DashboardHeader.tsx:51](https://github.com/is442oop/portfolio-analyzer-web-ui/blob/main/components/DashboardHeader.tsx#L51)

Reply:
thanks for pointing that out, made use of that so the display for the amount is more unified and standard!